src/libtriton/ast/ast.cpp

```diff
@@ -238,6 +243,7 @@ namespace triton {
       /* Init attributes */
       this->size = this->children[0]->getBitvectorSize();
       this->eval = ((this->children[0]->evaluate()) & this->getBitvectorMask());
+      this->symbolized = false;
```

Comment:
init() is used as an update function when replacing children in the simplification passes. Without a reset to false it can stay true even if all children become non symbolized.
I found the bug when simplifying (VAR - VAR) by 0..

Reply:
Oh nice catch! So, we have to add an test case in `./unittests/test_ast_simplification.py` to catch this bug in the future.

Reply:
I am realizing something, that to fix the bug I just described processSimplification needs to be fixed too since it uses setChild without calling init !
(I have my own processSimplification which uses init but it is part of another PR).
This is another reason to have a canReplaceNodeWithoutUpdate.. since updating parent nodes for each simplified child node is nuts... So temporarily I will also add a check in the next commit to not update the parent if all children nodes have not been simplified (original shared_pointer == returned shared_pointer), is that ok ?

Reply:
ok. Do it into the second PR then.